js/interface.js

```diff
@@ -124,14 +124,10 @@ var DynamicLists = (function() {
       }
     });
 
-    dataSourceProvider.then(function(dataSource) {
-      _this.config.dataSourceId = dataSource.data.id;
-    });
-  }
+    _this.config.dataSourceProvider = dataSourceProvider;
```

Comment:
I don't like we're adding to the `config` object content that isn't technically the widget configuration. It's being used here as a way to provide access to the provider promise in another file. There are many other ways to do this without unnecessarily changing the widget configuration.

Also, rather than triggering an event for an event listener to execute the intialisation, I'd rather we define a function to be executed.

See https://developers.fliplet.com/API/core/registry.html for how to use `Fliplet.Registry` to achieve this.

Examples:

- https://github.com/Fliplet/fliplet-widget-data-source/blob/487cb5f37007e3e3cc7d46f1e28edb4b26fc3623/js/preconfigured-rules.js
- https://github.com/Fliplet/fliplet-widget-notification-inbox/blob/8b44f8173578c95fc310f7e042b64dd7608ee5a9/js/app/libs.js
- https://github.com/Fliplet/fliplet-widget-dynamic-lists/blob/186709583e62c430c25d3fe8c369872dc59cf339/js/query-parser.js

Reply:
I added DSP by using `Fliplet.Registry` but I still need to trigger custom event. This event is not for executing initialization, it is used to notificate `interface-list` when to apply `then` for correct DSP saving. 
Otherwise we will need to wrap whole initialization in function to be executed, but this may cause more problems and take too much time.